### PR TITLE
Add coc.nvim language server configuration

### PR DESCRIPTION
**Pull request by klaaspieter**
Add vim editor configuration for [coc](https://github.com/neoclide/coc.nvim).

### CODE REVIEW

**Comment by benlangmuir** [comment]
Could you say a bit about how coc relates/compares with vim-lsp? Are there substantial advantages to one vs the other? Is it more/less popular?

**Comment by klaaspieter** [comment]
@benlangmuir Hard to say of course but I feel coc is definitely more popular within my bubble than vim-lsp. I hadn't heard of vim-lsp before I noticed it on this wiki actually.

Not much to go on but coc has (significantly) more forks, stars and watchers than vim-lsp...

File: Editors/README.md [hunk 1]
```diff
@@ -74,10 +74,13 @@ You will need the path to the `sourcekit-lsp` executable for the "command" secti
...
-## Vim 8
+## Vim 8 or Neovim
 
-Install [vim-lsp](https://github.com/prabirshrestha/vim-lsp). In your `.vimrc`, configure vim-lsp to use
-sourcekit-lsp for Swift source files like so:
+All methods below assume `sourcekit-lsp` is in your `PATH`. If it's not then replace `sourcekit-lsp` with the absolute path to the sourcekit-lsp executable.
+
+### vim-lsp
+
+In your `.vimrc`, configure vim-lsp to use sourcekit-lsp for Swift source files like so:
```
Comment (benlangmuir) [inline Editors/README.md]: The "Install vim-lsp" and link seems to have disappeared in this edit. Maybe these two sections should each contain a link to the underlying plugin - either vim-lsp or coc?

Comment (klaaspieter) [reply Editors/README.md]: It does look like I accidentally removed the link. Should be back now.

Comment (klaaspieter) [reply Editors/README.md]: Also added a link to coc.nvim :)

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test

**Pull request merged**